### PR TITLE
Add simple interoperability test with Cloudflare quiche

### PR DESCRIPTION
**Pull request by t8m**
This is an external test which requires recursive checkout of the cloudflare-quiche submodule.

We simply run a client against the example quiche-server serving HTTP/0.9 requests.

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/quic_client_test.c [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+{
+    int ec = SSL_get_error(s, ret);
+
+    return ec == SSL_ERROR_WANT_READ || ec == SSL_ERROR_WANT_WRITE;
+}
+
+static int test_quic_client(void)
+{
+    int testresult = 0, ret;
+    int c_fd = -1;
```
Comment (mattcaswell) [inline test/quic_client_test.c:31]: `int c_fd = INVALID_SOCKET;`

File: test/quic_client_test.c [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        SSL_tick(c_ssl);
+    }
+
+    testresult = 1;
+err:
+    SSL_free(c_ssl);
+    SSL_CTX_free(c_ctx);
+    BIO_ADDR_free(s_addr_);
+    BIO_free(c_net_bio_own);
+    if (c_fd >= 0)
```
Comment (mattcaswell) [inline test/quic_client_test.c:160]: `if (c_fd != INVALID_SOCKET)`

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: test/quic_client_test.c [hunk 3]
```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
```
Comment (paulidale) [inline test/quic_client_test.c:2]: 2023

**Review by paulidale — APPROVED** [review]
One nit.

**Comment by t8m** [comment]
@mattcaswell @paulidale fixup pushed

**Review by mattcaswell — APPROVED** [review]

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**